junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherFactory.java

```diff
@@ -26,6 +27,12 @@
  * of the implementation class of the {@link org.junit.platform.engine.TestEngine}
  * interface is declared.
  *
+ * <p>Test execution listeners are discovered accordingly. For that purpose, a text
```

Comment:
"accordingly" is insufficient here. 

Rather, we should mention that TELs which are discovered at runtime using the ServiceLoader mechanism will be automatically registered with the Launcher created by the factory but that clients of the Launcher may optionally register TELs manually in addition.

Not that you have to use my exact wording -- b/c I'd polish it myself -- but you get the gist.